### PR TITLE
Need To Replace kTLVType_Path With kTLVType_List Everywhere in the Code Fixes #4309

### LINKED ISSUE #4309: Need To Replace kTLVType_Path With kTLVType_List Everywhere in the Code
The spec was updated to rename CHIP TLV data type "Path" to "List". That should be updated everywhere in the code.

### PR DESCRIPTION
**Pull request by woody-apple**
#### Problem
Spec changed terms, updating code to match 

 #### Summary of Changes
Requested name changes

### CODE REVIEW

**Review by gerickson — APPROVED** [review]
/cc/ @holbrookt

**Comment by woody-apple** [comment]
@bzbarsky-apple @saurabhst @jelderton ?

**Review by saurabhst — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@andreilitvin @robszewczyk ?

**Review by andy31415 — APPROVED** [review]

File: src/lib/core/CHIPTLVDebug.cpp [hunk 1]
```diff
@@ -282,7 +282,7 @@ const char * DecodeType(const TLVType aType)
         retval = "Array";
         break;
 
-    case kTLVType_Path:
+    case kTLVType_List:
         retval = "Path";
```
Comment (emargolis) [inline src/lib/core/CHIPTLVDebug.cpp:286]: need to update as well...

**Pull request merged**